core/server/master/src/main/java/alluxio/master/file/meta/MutableInodeDirectory.java

```diff
@@ -209,7 +210,11 @@ public static MutableInodeDirectory fromJournalEntry(InodeDirectoryEntry entry)
     } else {
       ret.mDefaultAcl = new DefaultAccessControlList();
     }
-    ret.setMediumTypes(new HashSet<>(entry.getMediumTypeList()));
+    if (!entry.getMediumTypeList().isEmpty()) {
+      ret.setMediumTypes(new HashSet<>(entry.getMediumTypeList()));
+    } else {
+      ret.setMediumTypes(Collections.emptySet());
```

Comment:
Don't we already set the internal mediumtype list to `Collections.emptySet()` when it's empty?
Even if we not, we could be returning `Collections.emptySet()` from the `getMediumTypeList` call to avoid these if-else statements everywhere else.